README.md

```diff
@@ -1,7 +1,36 @@
-# virtual
+# Virtual
 
 Virtual method declaration.
 
+## Summary
+
+A _virtual method_ is a method which is inheritable and overridable. Since all methods in ruby are virtual, syntax is not required to provide this capability.
+ 
+When developing framework style code, it is often required to override or implement a specific method in order to satisfy an interface. The purpose of this library is to provide a way for a developer of framework code to convey that intent.
```

Comment:
Would be preferable to say, "in order to satisfy a contract". Though interfaces are a kind of contract, it's more proper to speak of this as a contract due to the lack of actual interface types. If this were a language with interface types, the statement would be fine, as the audience would have a better frame of reference (even though most Rubyists likely don't know what "contract" means, either).